core/src/main/java/org/mobilitydata/gtfsvalidator/notice/LeadingOrTrailingWhitespacesNotice.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.notice;
+
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * The value in CSV file has leading or trailing whitespaces.
+ *
+ * <p>This warning is emitted for values protected with double quotes since whitespaces for
+ * non-protected values are trimmed automatically by CSV parser.
+ *
+ * <p>Note that this is a warning: not an error because the feed may be still parsed and also not an
```

Comment:
Let's not become prescriptivists. If a ton of feeds are using those spaces, then our validator must accept them, so that may not be an error :)

This is like linguistics. If thousands of native speakers use a certain way of saying, it may not be proclaimed as an error. It may be a "lower style" (i.e., a warning in the case of validator) but not an error.

A nice video talking about "mistakes" that native English speaker make: https://www.youtube.com/watch?v=vGDb-fbvJmQ

And finally: the real way to fix that problem and avoid ambiguities is to give up CSV and switch static GTFS to protobufs. CSV is not a production data format, it is a toy format nice for office workers and students - everywhere where you do not care much about accuracy, speed and unambiguity.

Reply:
I'm against making discrepancies between the GTFS specification and the GTFS validator. If it is a common mistake and if it does not cause issues to GTFS consumers, then the rule
> Remove any extra spaces between fields or field names. Many parsers consider the spaces to be part of the value, which may cause errors.

should not be a requirement anymore, but one of the Best Practices.

Reply:
I like that you propose to amend the GTFS reference instead of enforcing it in its current shape.

Reply:
So what is our order of operations here @timMillet @carlfredl?

Is it a):
1. Merge this PR as a warning, even though the spec indicates it should be an error
2. Revise the spec to match the validator

...or b):
1. Modify and merge this PR as an error to match the current spec
2. Modify the spec to a treat as a warning (best practice instead of stated requirement)
3. Modify the validator to a warning to match the updated spec

Reply:
@aababilov I like to find a solution, as long as it is consistent :)

@barbeau I'd go with solution B. There is no guarantee that step 2 in both solutions will be completed, so going with option A will result in keeping a discrepancy between the spec and the validator.

Reply:
Thanks @timMillet!

>Which changes do you want me to make to this pull request before submit?

@aababilov So to merge this PR we just need the notice to be classified as an error instead of a warning.

Reply:
Done. I hope that we either fix the feeds or update the spec.